### ISSUE TITLE
Tsc doesn't typecheck

### ISSUE THREAD

**Issue by AliBasicCoder** [microsoft/TypeScript#42928]
# Bug Report

### 🕗 Version & Regression Information

```
❯ tsc --version
Version 4.1.5
```

### ⏯ Playground Link

[Playground link with relevant code](https://www.typescriptlang.org/play?#code/GYVwdgxgLglg9mABACxgCgJQC5EG0DOUATjGAOaIA+i4AJgKbCn20C6iA3gFCKJH1QQRJLjqNmbANxcAvly4QEhPIRLl2AXhToM0rqtJkAdACMYZTNKA)

### 💻 Code

```ts
function hi(): [string | undefined] {
  return [undefined];
}

const [string] = hi();

string.big();
```

### 🙁 Actual behavior

this code doesn't cause error when i run it with `tsc`

BUT this code DOES cause a error in the playground

### 🙂 Expected behavior

to error

**Comment by MartinJohns** [comment]
The playground has the `strictNullChecks` option enabled, which is by default disabled with the compiler.

**Comment by AliBasicCoder** [comment]
why `strictNullChecks` is disabled by default??

**Comment by MartinJohns** [comment]
To avoid breaking changes.
